toolz/tests/test_functoolz.py

```diff
@@ -508,3 +508,53 @@ def f(a, b):
         return a, b
 
     assert flip(f, 'a', 'b') == ('b', 'a')
+
+
+def test_excepts():
+    # These are descriptors, make sure this works correctly.
+    assert excepts.__name__ == 'excepts'
+    assert excepts.__doc__.startswith(
+        'A wrapper around a function to catch exceptions and\n'
+        '    dispatch to a handler.\n'
+    )
+
+    def idx(a):
+        """idx docstring
+        """
+        return [1, 2].index(a)
+
+    def handler(e):
+        """handler docstring
+        """
+        assert isinstance(e, ValueError)
+        return -1
+
+    excepting = excepts(ValueError, idx, handler)
+    assert excepting(1) == 0
+    assert excepting(2) == 1
+    assert excepting(3) == -1
+
+    assert excepting.__name__ == 'idx_excepting_ValueError'
+    assert 'idx docstring' in excepting.__doc__
+    assert 'ValueError' in excepting.__doc__
+    assert 'handler docstring' in excepting.__doc__
+
+    def getzero(a):
+        """getzero docstring
+        """
+        return a[0]
+
+    excepting = excepts((IndexError, KeyError), getzero)
+    assert excepting([]) is None
+    assert excepting([1]) == 1
+    assert excepting({}) is None
+    assert excepting({0: 1}) == 1
+
+    assert excepting.__name__ == 'getzero_excepting_IndexError_or_KeyError'
+    assert 'getzero docstring' in excepting.__doc__
+    assert 'return_none' in excepting.__doc__
+    assert 'Returns None' in excepting.__doc__
+
+    excepting = excepts(object(), object(), object())
+    assert excepting.__name__ == 'excepting'
+    assert excepting.__doc__ == excepts.__doc__
```

Comment:
It would be good to test that we only pass on the provided `Exception` type.  A small test that raises a different kind of error than the provided type would be useful.